cmd/renterd/config.go

```diff
@@ -225,39 +182,24 @@ func sanitizeConfig(cfg *config.Config) error {
 		cfg.Log.Database.Level = cfg.Log.Level
 	}
 
-	return nil
-}
+	// default data directory
+	cfg.Directory = defaultDataDirectory(cfg.Directory)
```

Comment:
while testing I noticed that this is not called before `setDataDirectory` so the data directory will actually not use the standard path on a fresh renter. Instead it will still use the local one.

Reply:
The other apps don't do this either. If you run a fresh renterd it will actually install in the default data directory of your OS, however that is if you don't go through the configuration wizard, if you do go through the wizard it will indeed install in the current directory. Perhaps we should change it so `setDataDirectory` defaults to the default data OS directory if `cfg.Directory` is not set.

Reply:
@n8maninger any thoughts? Considering that we should be consistent it affects `walletd` and `hostd` as well.
Imo it's a bit odd to try and do `renterd config` to generate a config just to have that config be in the current directory and if I switch to a different directory and run `renterd` it will use the defaults again.

e.g. when you run `rclone config` you'd also expect it to update the `~/.config/rclone/rclone.conf` file rather than creating a local one unless you overwrite it using some flag.

Reply:
I'm not sure I understand. If you run `hostd` or `walletd` config on a brand new node, it uses the default config and data directories. If this is not the behavior you're describing, please provide a reproduction in one of the other apps.

![Screenshot 2025-01-16 at 7 19 45 AM](https://github.com/user-attachments/assets/bf38d4dd-823d-4f19-8ac9-5206d247e251)
![Screenshot 2025-01-16 at 7 17 33 AM](https://github.com/user-attachments/assets/183a3632-ca88-475d-81b6-7cc4cde5475a)